TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+            }
+            
+            
+            NSString *token = [self->decodedJSONDictionary valueForKey:@"token"];
+            NSString *message = [self->decodedJSONDictionary valueForKey:@"message"];
+            
+            if (token == nil){
+                
+                NSString *errorMessage = @"User registration error";
+                
+                if (message !=  nil){
+                    
+                    errorMessage = message;
+                    
+                }
+                
+                [self uniNotification:@"Registration error" message:errorMessage notifName:@"authfailed" notifData:nil];
+                return;
+            }
+            
+            self->userInfo = [[NSMutableDictionary alloc] initWithDictionary:self->decodedJSONDictionary];
+            
+            appDelegate.token = token;
+            
+            [self uniNotification:@"Registration" message:@"Registration successful" notifName:@"authdone" notifData:nil];
+            
+        }else if (([data length] == 0) && (connectionError == 0)){
+            NSLog(@"result is empty");
+            
+            [self uniNotification:@"Connection error" message:@"Empty responce from server" notifName:@"authfailed" notifData:nil];
+            
+        }else if (connectionError != nil){
+            
+            NSLog(@"error happend: %@", connectionError);
+            
+            NSString *connectionErrorDescription = [connectionError description];
+            
+            [self uniNotification:@"Connection error" message:connectionErrorDescription notifName:@"authfailed" notifData:nil];
+            
+        }
+        
+    }];
+    
+    [postDataTask resume];
+    
+}
+
+
+#pragma mark - Notifications
+
+
+-(void)uniNotification:(NSString *)title message:(NSString *)message notifName:(NSString *)notifNname notifData:(NSDictionary *)notifData{
+    
+    //NS Log(@"cookie login ok");
+    
+    NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] init];
+    
+    [userInfo setValue:title forKey:@"title"];
+    [userInfo setValue:message forKey:@"message"];
+    
+    if (notifData != nil){
+        [userInfo setValue:notifData forKey:@"dictionary"];
+    }
+    
+    if (![NSThread isMainThread])
+    {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            
+            NSNotification *notification = [NSNotification notificationWithName:notifNname object:self userInfo:userInfo];
+            
+            [self sendNorification:notification];
+            
+        });
+        return;
+    }else{
+        
+        NSNotification *notification = [NSNotification notificationWithName:notifNname object:self userInfo:userInfo];
+        
+        [self sendNorification:notification];
+        
+    }
+    
+}
+
+
+
+-(void)sendNorification:(NSNotification *)notification
+{
+    
+    [[NSNotificationQueue defaultQueue]
+     enqueueNotification:notification
+     postingStyle:NSPostNow];
```

Comment:
this should be `NSPostASAP` and the above code could be simplified further then, I guess.